From c4acf2a2c333dba9292bada7073f02599330be49 Mon Sep 17 00:00:00 2001
From: Andrea Merello <andrea.merello@gmail.com>
Date: Wed, 25 Nov 2015 09:27:47 +0100
Subject: [PATCH 11/17] drm: introduce helper for accessing EDID blob in
 drm_connector structure

There is a bunch of drivers that directly access edid_blob_ptr in
drm_connector_structure. This requires a cast and possibly a NULL ptr
check.

Wrap this in a common helper.

Signed-off-by: Andrea Merello <andrea.merello@gmail.com>
Signed-off-by: Lars-Peter Clausen <lars@metafoo.de>

Conflicts:
	include/drm/drm_edid.h
---
 include/drm/drm_edid.h | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/include/drm/drm_edid.h b/include/drm/drm_edid.h
index 1e1908a..06f8521 100644
--- a/include/drm/drm_edid.h
+++ b/include/drm/drm_edid.h
@@ -25,6 +25,7 @@
 
 #include <linux/types.h>
 #include <linux/hdmi.h>
+#include <drm/drm_crtc.h>
 
 struct drm_device;
 struct i2c_adapter;
@@ -220,7 +221,6 @@ struct detailed_timing {
 #define DRM_EDID_YCBCR420_DC_MASK (DRM_EDID_YCBCR420_DC_48 | \
 				    DRM_EDID_YCBCR420_DC_36 | \
 				    DRM_EDID_YCBCR420_DC_30)
-
 /* ELD Header Block */
 #define DRM_ELD_HEADER_BLOCK_SIZE	4
 
@@ -429,6 +429,21 @@ static inline int drm_eld_size(const uint8_t *eld)
 }
 
 /**
+ * drm_connector_get_edid - Get current EDID from the given connector
+ * @connector: pointer to the connector stucture
+ *
+ * This is a helper for accessing the drm blob buffered in the connector
+ * struct (if any)
+ */
+static inline struct edid *drm_connector_get_edid(struct drm_connector *connector)
+{
+	if (!connector->edid_blob_ptr)
+		return NULL;
+
+	return (struct edid *)connector->edid_blob_ptr->data;
+}
+
+/**
  * drm_eld_get_spk_alloc - Get speaker allocation
  * @eld: pointer to an ELD memory structure
  *
-- 
1.9.1

